dashboard/app/controllers/api/v1/regional_partners_controller.rb

```diff
@@ -16,8 +16,41 @@ def capacity
     render json: {capacity: get_partner_cohort_capacity(regional_partner_value, role)}
   end
 
+  # GET /api/v1/pd/regional_partners/find
+  def find
+    if params[:school]
+      school = School.find(params[:school])
+      state = school.state
+      zip_code = school.zip
+    else
+      zip_code = params[:zip_code]
+      state = params[:state]
+
+      # lookup state abbreviation, since the supplied state can be the full name
+      state = get_us_state_abbr_from_name(state, true) if state && state.length > 2
+    end
+
+    # Find the matching partner
+    partner = RegionalPartner.find_by_region(zip_code, state)
+
+    if partner
+      render json: serialize_partner(partner)
+    else
+      render_404
+    end
+  end
+
   private
 
+  def serialize_partner(partner)
+    # The scope is not being passed to the serializer with `render json: serializer:` syntax,
+    # so initialize this explicitly.
+    # TODO (Andrew): Look into updating our very outdated version of ActiveModelSerializers
+    Api::V1::Pd::RegionalPartnerSerializer.new(
+      partner
+    ).attributes
+  end
```

Comment:
Looks like this was copied from [here](https://github.com/code-dot-org/code-dot-org/blob/c75b91f038fe55afcafd60770e8b8e6746c42c2c/dashboard/app/controllers/api/v1/pd/regional_partner_workshops_controller.rb#L37). It was a workaround to get the `scope` to work, but in this case we're not using scope so the comment doesn't make sense and this method is unnecessary.

Instead, line 37 can just be
```ruby
render json: partner, serializer: Api::V1::Pd::RegionalPartnerSerializer
```